.claude/scripts/agent-telemetry.sh

```
@@ -173,6 +175,33 @@ INJTMP=$(mktemp "${TMPDIR:-/tmp}/.agtel_inj.XXXXXXXX") || { echo "cannot create
+emit_injection_hits() {
+  local f="$1" session line raw record phrase
+  session=$(basename "$f" | tr -cd 'A-Za-z0-9._-')
+  [ -n "$session" ] || session=unknown
+
+  grep -niE "$INJ_PHRASE_RE" "$f" 2>/dev/null \
+    | while IFS=: read -r line raw; do
+        case "$line" in ''|*[!0-9]*) continue ;; esac
+        record=$(printf '%s' "$raw" | jq -r '.type // "malformed"' 2>/dev/null \
+                 | tr -cd 'A-Za-z0-9_-')
+        [ -n "$record" ] || record=malformed
+        printf '%s' "$raw" | grep -hoiE "$INJ_PHRASE_RE" | tr 'A-Z' 'a-z' \
```

Shellcheck (0.11.0)

[info] Use '[:upper:]' to support accents and foreign alphabets.

(SC2019)